src/scripts/main.js

```diff
@@ -1,3 +1,94 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('tbody');
+const container = document.querySelector('.container');
+const buttonAddRow = document.querySelector('.append-row.button');
+const buttonAddColumn = document.querySelector('.append-column.button');
+const buttonRemoveColumn = document.querySelector('.remove-column.button');
+const buttonRemoveRow = document.querySelector('.remove-row.button');
+
+const updateButtonStates = () => {
+  const firstRow = document.querySelector('tr');
+  const columns = firstRow ? firstRow.children.length : 0;
+  const rows = document.querySelectorAll('tr').length;
+
+  if (columns === 10) {
+    buttonAddColumn.setAttribute('disabled', 'disabled');
+  } else {
+    buttonAddColumn.removeAttribute('disabled');
+  }
+
+  if (columns <= 2) {
+    buttonRemoveColumn.setAttribute('disabled', 'disabled');
+  } else {
+    buttonRemoveColumn.removeAttribute('disabled');
+  }
+
+  if (rows === 10) {
+    buttonAddRow.setAttribute('disabled', 'disabled');
+  } else {
+    buttonAddRow.removeAttribute('disabled');
+  }
+
+  if (rows <= 2) {
+    buttonRemoveRow.setAttribute('disabled', 'disabled');
+  } else {
+    buttonRemoveRow.removeAttribute('disabled');
+  }
+};
+
+container.addEventListener('click', (e) => {
+  if (e.target.matches('.button')) {
+    updateButtonStates();
+  }
```

Comment:
The event listener on the container is checking if the clicked target matches the '.button' class and then calls `updateButtonStates()`. Ensure that this logic aligns with the intended functionality, as it might be redundant if each button already updates the state after its specific action.